autonomy/chain/tx.py

```diff
@@ -193,13 +193,27 @@ def transact(
                 if not should_retry(error):
                     raise ChainInteractionError(error) from e
                 if should_reprice(error):
-                    print("Repricing the transaction...")
+                    print(f"Low gas error: {e}; Repricing the transaction...")
                     tx_dict = self._reprice(cast(Dict, tx_dict))
-                    continue
+                    continue  # pragma: nocover
+                if "Transaction with hash" in error and "not found" in error:
+                    already_known = True
```

Comment:
If we've sent a tx and cannot get receipt from the tx hash we retry instead of sending a new transaction